src/components/slider/slider.ts

```diff
@@ -215,6 +228,80 @@ export class MdSlider implements AfterContentInit {
   }
 
   /**
+   * Calls the appropriate method for calculating the tick separation depending on what the tick
+   * interval is. If there is no tick interval or the interval is set to something other than a
+   * number or 'auto', nothing happens.
+   */
+  private _calculateTickSeparation() {
+    if (this._tickInterval == 'auto') {
+      this._calculateAutoTickSeparation();
+    } else if (Number(this._tickInterval)) {
+      this._calculateTickSeparationFromInterval();
+    }
+  }
+
+  /**
+   * Calculates the separation in pixels of tick marks by starting with the assumption every step
+   * needs a tick and eliminating the number of ticks until there is a distance of at least 30px
+   * between each tick.
+   */
+  private _calculateAutoTickSeparation() {
+    // The pixel value for how far apart the ticks should be.
+    let tickSeparation = 0;
+    // Keeps track of how many steps to multiply the slider's step by.
+    let stepCounter = 1;
+
+    while (tickSeparation < MIN_AUTO_TICK_SEPARATION) {
```

Comment:
So, you don't actually need this loop; you can calculate the appropriate multiple of `step`. This would look something like
```ts
stepCounter = Math.ceil(MIN_AUTO_TICK_SEPARATION * (max - min) / width * step)
```
(though obviously it's not a "counter" any more)

[working it out](https://goo.gl/photos/qyjjBkR5QG8YG9P79)